util/src/main/java/com/google/cloud/hadoop/util/RetryHttpInitializer.java

```diff
@@ -295,6 +302,12 @@ public void initialize(HttpRequest request) {
           "Request is missing a user-agent, adding default value of '%s'", defaultUserAgent);
       request.getHeaders().setUserAgent(defaultUserAgent);
     }
+
+    if (headers != null) {
+      for (Map.Entry<String, String> header : headers.entrySet()) {
+        request.getHeaders().set(header.getKey(), header.getValue());
```

Comment:
Does it have `setAll` or similar method that you can use here instead of loop?

Reply:
Looked up and doesn't seem like HttpHeaders class has such API.